### PR TITLE
Adding the embargo check in full text update.

### PR DESCRIPTION
**Pull request by knjackson30974**

### CODE REVIEW

File: lib4ridora.module [hunk 1]
```diff
@@ -1419,3 +1416,18 @@ function lib4ridora_ir_citationCModel_islandora_object_ingested($object) {
   module_load_include('inc', 'lib4ridora', 'includes/full_text');
   lib4ridora_update_full_text($object);
 }
+
+/**
+ * Implements hook_cmodel_pid_islandora_datastream_modified().
+ */
+function lib4ridora_ir_citationCModel_islandora_datastream_modified($object, $datastream) {
```
Comment (qadan) [inline lib4ridora.module]: Should we check if we're modifying or ingesting a PDF datastream so we don't create versions when not necessary? We set those mimetypes to [application/pdf](https://github.com/knjackson30974/lib4ridora/blob/lib-115/includes/pdf-upload.form.inc#L143); could we check against that?

Comment (adam-vessey) [inline lib4ridora.module]: This hook doesn't appear to exist... We only use "model_dsid", no "model" alone: https://github.com/Islandora/islandora/blob/7.x/includes/tuque_wrapper.inc#L71-L74

... [`islandora.api.php`](https://github.com/Islandora/islandora/blob/7.x/islandora.api.php#L346-L353) should probably be modified... Could just roll another ticket for some other time?

File: lib4ridora.module [hunk 2]
```diff
@@ -1419,3 +1416,20 @@ function lib4ridora_ir_citationCModel_islandora_object_ingested($object) {
...
+  module_load_include('inc', 'lib4ridora', 'includes/full_text');
+  if ($datastream->mimetype == 'application/pdf') {
+    lib4ridora_update_full_text($object);
+  }
+}
+/**
+ * Implements hook_cmodel_pid_dsid_islandora_datastream_ingested().
+ */
+function lib4ridora_ir_citationCModel_PDF_islandora_datastream_ingested($object, $datastream) {
```
Comment (adam-vessey) [inline lib4ridora.module]: Also... to deal with the ["other" PDF datastreams](https://github.com/discoverygarden/lib4ridora/blob/7.x/includes/embargo.form.inc#L32-L49), may need to implement the base `hook_islandora_datastream_ingested()` variant, and check the model and dsid ourselves.

File: lib4ridora.module (hunk 1 above)
Comment (qadan) [inline lib4ridora.module]: ... _should_ we be using "model"? Would be a one-line alter to `islandora_invoke_datastream_hooks`, but would potentially add a lot of extra refinements to the hook list?

Comment (adam-vessey) [inline lib4ridora.module]: I'm not so sure it's useful, but not really opposed... When necessary, can use the base `hook_islandora_datastream_modified` and take the filtering into our own hands.

... So... shrug?

**Pull request merged**